src/protocol.rs

```diff
@@ -246,6 +231,11 @@ impl SignatureShare {
         }
     }
 
+    /// Create a new [`SignatureShare`] given (r, s).
+    pub fn new_sig(r: Option<k256::Scalar>, s: k256::Scalar) -> Self {
```

Comment:
question: does this need to be public or is `pub(crate)` enough?

Also, the canonical name in Rust for a constructor like this is just plain `new`.

Reply:
Oh, I missed that. @amaloz flagged that empty constructor in #356 as something that should go away; for now I think it would be better to paste the implementation of the empty `new` function into the `default` function and use the name `new` for the actual constructor you just created.

Reply:
Thanks for your comments @marsella! All of them are addressed. Ready for re-review!